ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/FastSyncDownloader.java

```diff
@@ -85,6 +85,7 @@ protected CompletableFuture<FastSyncState> start(final FastSyncState fastSyncSta
     if (worldStateStorage instanceof BonsaiWorldStateKeyValueStorage) {
       LOG.info("Clearing bonsai flat account db");
       worldStateStorage.clearFlatDatabase();
+      worldStateStorage.clearTrieLog();
```

Comment:
why do we need to do this?

Reply:
we will not be able to go back before this pivot block. In addition, we could have bad trielog so it is preferable to heal them too. Finally keeping old trielog could make a false positive for the isWorldStateAvailable.